### PR TITLE
Feature/refactor log

### PR DESCRIPTION
**Pull request by mariofv**
What's new:
- Refactored Log system. 
- Logs are printed to Window console, Visual studio console, Engine console and `Game/logs/log.txt`.
- Scripts can log now using `App->engine_log->Log("Whatever %s", "cool!")` (no need of aditional `#include`).
- Added new debug log to debug purposes. Use `DEBUG_LOG`.

### CODE REVIEW

File: Engine/Filesystem/Path.cpp [hunk 1]
```diff
@@ -250,9 +250,20 @@ std::string Path::GetFullPath() const
...
 
+std::string Path::GetFilename(const std::string& path)
+{
+	return path.substr(path.find_last_of('/') + 1, -1);
+}
+
+std::string Path::GetFilenameWindows(const std::string& path)
+{
+	return path.substr(path.find_last_of('\\') + 1, -1);
+}
```
Comment (Dartemiss) [inline Engine/Filesystem/Path.cpp:261]: Functions should be const right?

File: Engine/Log/Log.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#include "Main/Globals.h"
+#include "Main/Application.h"
+#include "Filesystem/Path.h"
+#include "Log/EngineLog.h"
+#include <mutex>
+
+void LittleOrionLogEntry(const EngineLog::LogEntryType type, const char file[], int line, const char* format, ...)
```
Comment (Dartemiss) [inline Engine/Log/Log.cpp:7]: Just asking, what does it means to put ... as parameter?

**Review by Dartemiss — APPROVED** [review]
GJ man!

imagen

File: Engine/Filesystem/Path.cpp (hunk 1 above)
Comment (mariofv) [reply Engine/Filesystem/Path.cpp:261]: Because they are static they cannot be const

File: Engine/Log/Log.cpp (hunk 2 above)
Comment (mariofv) [reply Engine/Log/Log.cpp:7]: It's varying list of arguments, it means that you can pass any number of parameters after format

File: Engine/Filesystem/Path.cpp (hunk 1 above)
Comment (Dartemiss) [reply Engine/Filesystem/Path.cpp:261]: my bad didnt realize

**Review by JorxSS — APPROVED** [review]
Good Job LP.

image

**Pull request merged**